src/lib/yacc.rs

```diff
@@ -216,18 +239,23 @@ impl YaccParser {
                 }
 
             },
-            None => Err(YaccError{kind: YaccErrorKind::IllegalString, off: i})
+            None => {
+                Err(self.mk_error(YaccErrorKind::IllegalString, i))
+            }
         }
     }
 
-    fn parse_ws(&self, i: usize) -> YaccResult<usize> {
-        match RE_WHITESPACE.find(&self.src[i..]) {
-            Some((s, e)) => {
-                assert!(s == 0);
-                Ok(i + e)
+    fn parse_ws(&mut self, i: usize) -> YaccResult<usize> {
```

Comment:
Agreed that it's more consistent. However I think it would be "cleaner" to do it differently for two reasons:

* The fact that ``parse_ws`` can't fail would be part of the function signature.
* Every time ``parse_ws`` is used, there is currently a (sort of) redundant ``try!`` (e.g. ``i = try!(self.parse_ws(k))``)